### ISSUE TITLE
Weird behaviour with iphone's home bar

### ISSUE THREAD

**Issue by srkuleo** [emilkowalski/vaul#199]
Firstly I made a simple drawer with RadixUI, but it didn't have a smooth sliding animation and didn't feel that much native. Decided to swap for Vaul drawer, simply replacing regular Dialog elements from Radix with Drawer elements from Vaul.

In theory, it should work and it does to some extent, but somehow it starts few px's above the Home bar section and slides back to the same point, then "footer" weridly disappears after closing. Wasn't encountering this bug with first version of component.

Here is the code for the component: 

```
import type { Workout } from "./Workouts";
import { Drawer } from "vaul";
import { PreviewIcon } from "@/icons/user/preview";

export const AltPreview = ({ workout }: { workout: Workout }) => {
  return (
    <Drawer.Root shouldScaleBackground>
      <Drawer.Trigger className="rounded-lg px-2 py-1 shadow-md ring-1 ring-inset ring-slate-300 transition focus:outline-none active:scale-95 dark:shadow-slate-900 dark:ring-slate-600">
        {PreviewIcon}
      </Drawer.Trigger>

      <Drawer.Portal>
        <Drawer.Overlay className="fixed inset-0 bg-slate-900/40 backdrop-blur-xs dark:bg-slate-950/70" />

        <Drawer.Content className="fixed inset-0 mt-32 rounded-t-modal bg-white dark:bg-slate-800">
          <div className="flex items-center justify-between p-4">
            <p className="text-lg font-bold">
              Current workout: {workout.title}
            </p>
            <Drawer.Close className="font-extrabold text-violet-500 focus:outline-none active:text-violet-300 dark:text-violet-400 active:dark:text-violet-600">
              Done
            </Drawer.Close>
          </div>
        </Drawer.Content>
      </Drawer.Portal>
    </Drawer.Root>
  );
};
```

and a video showcasing weird behaviour.

https://github.com/emilkowalski/vaul/assets/91840410/b1aee044-4c95-475f-8d40-a18f69fd441d

**Comment by emilkowalski** [comment]
Is this PWA?

**Comment by srkuleo** [comment]
Yes, it is. I somehow menaged to barely fix it by adding w-screen class on html element, but sometimes after logging in, it returns and shows weird outline after opening (probably Radix's thing).

**Comment by emilkowalski** [comment]
This code ensures that the Safari toolbar shows up when the drawer is opened. Otherwise, if you had the toolbar on the bottom on iOS you would experience a delayed shift after the toolbar was opened. Considering there is no toolbar when you use it in PWA mode we can disable that via a prop, I will add it soon. Thanks for the report!

**Comment by emilkowalski** [comment]
I'm actually unable to reproduce, could you create a codesandbox demo with the code that causes this in a PWA env?

https://github.com/emilkowalski/vaul/assets/36730035/22a911f3-4b6b-42a1-a9cc-72c9f64a5a2d

**Comment by srkuleo** [comment]
Sorry for the delayed respond.  

I am using Kinde as Auth provider and only time I am encountering this bug is after user open app while previously being authenticate. In these videos below you can see 3 different cases. 

1st: User automatically being redirected to Home page after opening app without calling an auth API. No bug occuring.


https://github.com/emilkowalski/vaul/assets/91840410/7cfe8325-42bf-4dbb-8126-20928d20c525


2nd: User asked to authenticate via auth API. No bug occuring.


https://github.com/emilkowalski/vaul/assets/91840410/7ad0cb04-846f-49a3-8893-886b0d90f284


3rd: Kind of weird case, since user isn't immediately redirected to Home page after opening app, auth API gets prompted but for really short time (probably confirms that user is authed), then redirects to Home page. Still don't know why it occurs, probably some weird PWA behaviour. In this case bug is occuring.


https://github.com/emilkowalski/vaul/assets/91840410/ddb8064f-adad-4d99-a85f-814a9d7874a2


In my opinion, the easiest way to reproduce is by logging in at https://noteset.vercel.app/ or cloning my repo https://github.com/srkuleo/noteset and providing required env variables. I am sorry if I am asking too much, if it's really needed I will provide codesandbox demo.

**Comment by emilkowalski** [comment]
Fixed in #277
